### DISCUSSION TITLE
Open Simplex Noise question

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#134] [Q&A] [answered]
I was studying the Numpy Vectorization tutorial, when I use `os_noise()` instead of `noise()`  the third argument seems to be ignored... 
What am I doing wrong? Could it be failing for not being a numpy array like the others? I tried to fill a third np_array with the values but it didn't help.

![image](https://user-images.githubusercontent.com/3694604/185807006-fc2798c8-36a9-4062-b456-2fa55961a2aa.png)

```python
import py5
import numpy as np

s = 0.005
ygrid, xgrid = np.mgrid[0:500:10, 0:500:10]


def setup():
    py5.size(500, 500, py5.P3D)
    py5.noise_seed(0)
    py5.color_mode(py5.HSB)


def draw():
    py5.background(0)
    py5.lights()
    py5.translate(py5.width / 2, py5.height / 2, py5.height / 2)
    py5.rotate_x(py5.QUARTER_PI)
    py5.translate(-py5.width / 2, -py5.height / 2, 0)
#    d = 128 + 128 * py5.os_noise(frame_count * s, xgrid * s, ygrid * s)  # try for compariisson...
    d = 128 + 128 * py5.os_noise(xgrid * s, ygrid * s, py5.frame_count * s)
    my_box(xgrid, ygrid, -200 + d / 2, 10, 10, d)
    print(py5.frame_count)


@np.vectorize
def my_box(x, y, z, *args):
    py5.push()
    py5.translate(x, y, z)
    py5.fill(args[-1] % 256, 200, 200)
    py5.box(*args)
    py5.pop()


py5.run_sketch()

```

**Comment by hx2A** [comment, marked as answer]
This is a bug, caused by a stupid typo. Good catch!

BTW, I think you want `os_noise_seed()` if you are using `os_noise()`. I'm glad you gave vectorization a try here, both because it is faster and because it uncovered this bug.

**Reply by hx2A** [reply]
The fix has been pushed to main. Thanks!
